lib/ansible/modules/windows/win_domain_membership.ps1

```diff
@@ -168,15 +170,15 @@ Function Join-Workgroup {
 
         # 2012+ call the Workgroup arg WorkgroupName, but seem to accept
         try {
-            $rc_result = Remove-Computer -Workgroup $workgroup_name -Credential $domain_cred -Force
+            Remove-Computer -Workgroup $workgroup_name -Credential $domain_cred -Force
         } catch {
             Fail-Json -obj $result -message "failed to remove computer from domain: $($_.Exception.Message)"
         }
     }
 
     # we're already on a workgroup- change it.
     Else {
-        $swg_result = Set-Workgroup $workgroup_name
+        Set-Workgroup $workgroup_name
```

Comment:
My comment still stands here, `Set-Workgroup` returns something and we should be redirecting it to `$null` to no confuse the json parser on the controller side.

Reply:
and my response too :), changing it breaks the CI - you want to keep it or should I look into changing the tests?

Reply:
Hmm I don't see how based on looking at the code, do you have a link to the failure? I'm just looking at `Set-Workgroup` and the return statement is on a newline but still part of the failing brace so it doesn't even look like that does anything https://github.com/ansible/ansible/blob/66e0041b3ab5d24ae51f28938cf6a90af60374e6/lib/ansible/modules/windows/win_domain_membership.ps1#L158. So maybe just return that line from the function as the module will exit out when calling `Fail-Json`.

Reply:
https://app.shippable.com/github/ansible/ansible/runs/125201

Reply:
Ahhh, it's because we call `Fail-Json` which outputs the return json in the output stream instead of stdout. While we exit it early the json string is redirected. Ok If you can just fix up the indentation of line 158 and remove that return statement I think we are all good.

Reply:
you want the function to not return anything then?

Reply:
Yes, the `Set-Workgroup` function has the following line https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_domain_membership.ps1#L156. It makes it look like the function returns that result but because the brace for the if statement is placed after this call it is actually being run after `Fail-Json`. `Fail-Json` stops the pipeline so nothing will ever be called making the `return` line unreachable. So right now it works it's just horribly confusing and leads to the conversation above.

Reply:
removed, let's see what happens - fingers crossed